src/sonic_ax_impl/mibs/ietf/rfc2737.py

```diff
@@ -35,76 +110,256 @@ def __init__(self):
+        self.physical_hw_version_map = {}
+        self.physical_serial_number_map = {}
+        self.physical_mfg_name_map = {}
+        self.physical_model_name_map = {}
 
     def reinit_data(self):
         """
         Re-initialize all data.
         """
-        device_metadata = self.statedb.get_all(self.statedb.STATE_DB, self.DEVICE_METADATA)
 
-        # TODO: Add support for unicode in swsssdk
-        chassis_serial_number = ""
+        # reinit cache
+        self.physical_classes_map = {}
+        self.physical_description_map = {}
+        self.physical_hw_version_map = {}
+        self.physical_serial_number_map = {}
+        self.physical_mfg_name_map = {}
+        self.physical_model_name_map = {}
+
+        device_metadata = mibs.get_device_metadata(self.statedb)
+        chassis_sub_id = (self.CHASSIS_ID, )
+        self.physical_entities = [chassis_sub_id]
+
         if not device_metadata or not device_metadata.get(b"chassis_serial_number"):
             chassis_serial_number = ""
         else:
             chassis_serial_number = device_metadata[b"chassis_serial_number"]
-            self.physical_classes = [(self.CHASSIS_ID, )]
-            self.physical_classes_map = {
-                    (self.CHASSIS_ID, ): (PhysicalClass.CHASSIS, chassis_serial_number.decode("utf-8"))
-                }
 
+        self.physical_classes_map[chassis_sub_id] = PhysicalClass.CHASSIS
+        self.physical_serial_number_map[chassis_sub_id] = chassis_serial_number
+
+        # update interface maps
+        _, self.if_alias_map, _, _, _ = \
+            mibs.init_sync_d_interface_tables(SonicV2Connector())
+            
     def update_data(self):
         """
         Update cache.
-        NOTE: Nothing to update right now. Implementation is required by framework.
         """
-        pass
+
+        chassis_sub_id = (self.CHASSIS_ID, )
+        self.physical_entities = [chassis_sub_id]
+
+        transceiver_key_pattern = mibs.transceiver_info_table("*")
+        self.transceiver_entries = self.statedb.keys(self.statedb.STATE_DB,
+                                                     transceiver_key_pattern)
+
+        if self.transceiver_entries is None:
+            # nothing in DB
+            return
+
+        # decode keys from DB
+        transceiver_entries = map(lambda k: k.decode("utf-8"), self.transceiver_entries)
+        # update xcvr info from DB
+        # use port's name as key for transceiver info entries
+        for transceiver_entry in transceiver_entries:
+            # extract interface name
+            interface = transceiver_entry.split("|")[-1]
+
+            ifalias = self.if_alias_map.get(interface.encode("utf-8"), b"")
+            # decode alias
+            ifalias = ifalias.decode("utf-8")
+
+            # get interface from interface name
+            ifindex = port_util.get_index_from_str(interface)
+
+            if ifindex is None:
+                # interface name invalid, skip this entry
```

Comment:
Shall we log an error?

Reply:
Added a warning message, since we can skip the incorrect entry and continue.